CONTRIBUTING.md

```diff
@@ -96,6 +96,155 @@ pnpm scripts create-secrets -e next
+
+ 4. Select **#releases** as the channel the webhook posts to
+
+ 5. Select **Copy Webhook URL**
+
+ 6. Add the url to your `.env` file as `DISCORD_WEBHOOK_URL_RELEASES`
+
+<img src=./docs/guide/github-releases-webhook.png width="900">
+
+### Creating the GitHub Webhook
+
+Go to either the GitHub organization or repository that you wish to recieve release notifications for. You must be the owner of the organization or repository to perform the following steps.
+
+ 1. Go to the desired GitHub organization/repository, and click **Settings**
+
+ 2. Click **Webhooks** -> **Add Webhook**
+
+ 3. Under **Payload URL** , paste in the Discord Webhook URL from before with `/github` appended to the end
+
+ 4. Under **Content type**, select `application/json`
+
+ 5. Create a **Secret** and store it in `.env` as `GITHUB_RELEASES_WEBHOOK_SECRET`
+
+ 6. Select  **Let me select individual events** then choose `releases`
+
+ 7. Click **Add Webhook**
+
+<img src=./docs/guide/github-webhook.png width="900" >
+
+### Configuring an organization membership webhook
+
+Configuring this webhook will remove the Discord `staff` role when members leave your organization.
+
+#### Create the GitHub webhook
+
+You must be the owner of the GitHub organization to perform the following steps.
+
+1. Go to your GitHub organization, and click **Settings** -> **Webhooks** -> **Add Webhook**
+
+2. Under **Payload URL**  choose `http://<your-production-url>/api/webhooks/github-org-membership`
+
+3. Under **Content type**, select `application/json`
+
+4. Create a **Secret** and store in in `.env` as `GITHUB_ORG_WEBHOOK_SECRET`
+
+5. Select  **Let me select individual events** then choose `Organizations`
+
+6. **Add Webhook**
+
+### Configuring a GitHub App
+
+A GitHub App is required to obtain the permissions necessary for many API calls. You must be the owner of the GitHub organization to perform the following steps.
+
+#### Creating the App
+
+1. Go to your GitHub organization, and click **Settings** -> **Developer Settings** -> **GitHub Apps**
+
+2. Select **New GitHub App**
+
+3. For the homepage URL, put in the homepage of your website
+
+4. Under **Callback URL** put `http://<your-production-url>/api/auth/callback/github`
+
+5. Select  **Expire user authorization tokens** AND **Request user authorization (OAuth) during installation**
+
+6. Under **Webhook** deselect **Active**
+
+7. Scroll down to **Organization Permissions** and under **Members** select `Read-only`
+
+8. **Create GitHub App**
+
+<img src=./docs/guide/github-app-permissions.png width="900" >
+
+### Storing App IDs
+
+Now you should be looking at the settings page for your app.
+
+<img src=./docs/guide/github-app-ids.png width="900" >
+
+1. Copy the App ID and store in `.env` as `GITHUB_APP_ID`
+
+2. Copy the Client ID and store in `.env` as `GITHUB_CLIENT_ID`
+
+3. Select **Generate a new client secret**, then copy and store in `.env` as `GITHUB_CLIENT_SECRET`
+
+## Generating a Private Key
```

Comment:
are these meant to be h4's? (nested underneath Configuring the GitHub App)

Reply:
yes just changed

Reply:
Nice! LGTM